### ISSUE TITLE
Trying to use taglib in rust ith unicode characters in path

### ISSUE THREAD

**Issue by SirWindfield** [chyyran/seiri#12]
I am working on a project that needs to parse music files that have unicode characters in their file path. I have some questions for you if you don't mind:

1) Did you actually solve this issue or does seiri not support unicode file paths?
2) If you solved it, how did you do it? 

I took a look at `katatsuki` but it didn't help me a lot. It looks like you wrap the C++ version of taglib2 (I tried to use `taglib-rust` for my program but got some errors when trying to open uncidoe file paths) into a more general C++ class that handles opening/closing files and returns metadata. Then you wrap that part around c and actually use the C part when building the library. Which would actually mean that seiri shouldn't support unicode characters because the pointer type passed to the `create_tracl_data` method in `lib.rs` is `*const ::std::os::raw::c_char`, a `c_char` pointer which is usually ascii only.

Would be super nice of you if you could help me out! 
Have a great weekend~

**Comment by chyyran** [comment]
`katatsuki` does indeed support UTF-8 path names on both Windows and Linux.

If you look into `TrackData.cpp`, it creates a `TagLib::String` from the
raw C-style string using the `*const c_char` pointer passed into
`create_track_data`. The `TagLib::String` (which is owned) is used instead
of the raw string when reading the tags.

The `*const c_char` comes from `std::String::into_raw()`, thus is not an
ASCII string, but contains UTF-8 bytes as do all Rust strings.
`TagLib::String` then converts these UTF-8 bytes to a `wchar_t` backed
`TagLib::FileName` on Windows or leaves them alone on other platforms.

**Comment by SirWindfield** [comment]
Thanks! I moved from `taglib-rust` to `katatsuki` for the time being. It works perfectly. I had problems with `taglib-rust` that it won't work with paths that contain special unicode characters (looks like it is more of an edge case tbh). The following characters didn't work: `ß`, `Ä`. 
I also tested `taglib-rust` with other unicode chars (I loaded up some of my japanese podcasts) and those worked perfectly fine. Kinda strange.... :smile:

Thanks for your write-up though!
